app.rb

```diff
@@ -0,0 +1,125 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'capitalize_decorator'
+require_relative 'trimmer_decorator'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts "List of Books (#{@books.length}):"
+    @books.each do |book|
+      puts "#{book.id} - #{book.title} by #{book.author}"
+    end
+  end
+
+  def list_people
+    puts "List of People (#{@people.length}):"
+    @people.each do |person|
+      puts "#{person.id} - #{person.name} (#{person.is_a?(Student) ? 'Student' : 'Teacher'})"
+    end
+  end
+
+  def create_person
+    puts "Choose the type of person:\n1. Student\n2. Teacher"
+    print 'Enter the type (1 or 2): '
+    type = gets.chomp.to_i
+
+    if type == 1
+      student = create_student
+      @people << student unless student.nil?
+      puts "Student #{student.name} created."
+    elsif type == 2
+      teacher = create_teacher
+      @people << teacher unless teacher.nil?
+      puts "Teacher #{teacher.name} created."
+    else
+      puts 'Invalid choice. No person created.'
+    end
+  end
+
+  def create_student()
+    print 'Name: '
+    name = gets.chomp
+    print 'Classroom: '
+    classroom = gets.chomp.to_str
+    return nil unless classroom.length >= 1
+
+    print 'Age (numeric): '
+    age = gets.chomp.to_i
+    return nil unless age.positive?
+
+    print 'Parent permission (y for true or n for false): '
+    parent_permission = gets.chomp
+
+    Student.new(classroom, age, name: name, parent_permission: parent_permission.downcase == 'y')
+  end
+
+  def create_teacher()
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    return nil unless specialization.length >= 1
+
+    print 'Age (numeric): '
+    age = gets.chomp.to_i
+    return nil unless age.positive?
+
+    print 'Parent permission (y for true or n for false): '
+    parent_permission = gets.chomp
```

Comment:
- [ ] Please remove Parent permission because the teacher don't want parent permission the student want parent permission only so kindly fix this issue as well :slightly_smiling_face:

Reply:
Thanks for pointing this out. I never wanted to do that but a reviewer from previous reviews insisted on it.